File: docs/prd/voting-v2/_capture.mjs
import { chromium } from 'playwright';
import { mkdir } from 'node:fs/promises';
import { dirname, join } from 'node:path';
import { fileURLToPath } from 'node:url';

const __dirname = dirname(fileURLToPath(import.meta.url));
const out = join(__dirname, 'screenshots');
const BASE = 'http://localhost:5173';

async function shot(page, name, locator) {
  const path = join(out, `${name}.png`);
  try {
    if (locator) {
      const el = locator.first();
      if (await el.isVisible({ timeout: 2500 })) {
        await el.screenshot({ path, timeout: 8000 });
        console.log('ok', name);
        return;
      }
    }
  } catch (err) {
    console.warn('fallback-full', name, String(err.message || err).slice(0, 120));
  }
  await page.screenshot({ path, fullPage: true });
  console.log('page', name);
}

async function gotoHash(page, hash, waitText) {
  await page.goto(`${BASE}/${hash}`, { waitUntil: 'networkidle' });
  if (waitText) {
    await page.getByText(waitText).first().waitFor({ timeout: 20000 }).catch(() => {});
  }
  await page.waitForTimeout(400);
}

const browser = await chromium.launch({ headless: true, channel: 'chrome' });
await mkdir(out, { recursive: true });

const admin = await browser.newContext({ viewport: { width: 1440, height: 900 }, deviceScaleFactor: 1 });
const ap = await admin.newPage();
ap.setDefaultTimeout(20000);

await gotoHash(ap, '#/voting-v2/vote-v2-overview', '概览');
await shot(ap, '投票概览-页面整体');
await shot(ap, '投票应用-左侧菜单', ap.locator('.ant-layout-sider').first());
await shot(ap, '投票概览-KPI指标卡', ap.locator('.overview-kpi-grid').first());
await shot(ap, '投票概览-进行中表格', ap.locator('.ant-card').filter({ hasText: '进行中的投票' }).first());

await gotoHash(ap, '#/voting-v2/vote-v2-layout', '装修');
await shot(ap, '投票装修-移动端工作台');
await shot(ap, '投票装修-组件面板', ap.locator('.deco-palette, [class*="palette"]').first());

await gotoHash(ap, '#/voting-v2/vote-v2-layout-pc', '装修');
await shot(ap, '投票装修-PC工作台');

await gotoHash(ap, '#/voting-v2/vote-v2-list', '投票管理');
await shot(ap, '投票管理-页面整体');
await shot(ap, '投票管理-查询筛选', ap.locator('.search-card').first());
await shot(ap, '投票管理-查询筛选区', ap.locator('.search-card').first());
await shot(ap, '投票管理-列表表格', ap.locator('.ant-table').first());
await shot(ap, '投票管理-工具栏新增', ap.getByRole('button', { name: '新增' }).first());

async function openRowMore(page, campaignName) {
  await page.getByRole('button', { name: new RegExp(`更多操作 ${campaignName}`) }).click({ force: true });
  await page.waitForTimeout(250);
}

try {
  await openRowMore(ap, '部门十佳员工评选');
  await ap.getByRole('menuitem', { name: '分享' }).click({ force: true });
  await ap.getByRole('dialog').waitFor({ timeout: 8000 });
  await shot(ap, '投票管理-分享弹窗', ap.getByRole('dialog'));
  await ap.keyboard.press('Escape');
  await ap.getByRole('dialog').waitFor({ state: 'hidden', timeout: 5000 }).catch(() => {});
} catch (err) {
  console.warn('share-modal', err.message);
  await shot(ap, '投票管理-分享弹窗');
}

try {
  await openRowMore(ap, '车间安全之星');
  await ap.getByRole('menuitem', { name: '删除' }).hover();
  await ap.waitForTimeout(500);
  await shot(ap, '投票管理-进行中删除禁用Tooltip', ap.locator('.ant-tooltip:visible').first());
  await ap.keyboard.press('Escape');
  await ap.mouse.move(0, 0);
} catch (err) {
  console.warn('delete-tooltip', err.message);
  await shot(ap, '投票管理-进行中删除禁用Tooltip');
}

try {
  await openRowMore(ap, '部门十佳员工评选');
  await ap.getByRole('menuitem', { name: '删除' }).click({ force: true });
  await ap.getByRole('dialog').waitFor({ timeout: 8000 });
  await shot(ap, '投票管理-删除确认弹窗', ap.getByRole('dialog').last());
  await ap.keyboard.press('Escape');
  await ap.getByRole('dialog').waitFor({ state: 'hidden', timeout: 5000 }).catch(() => {});
} catch (err) {
  console.warn('delete-confirm', err.message);
  await shot(ap, '投票管理-删除确认弹窗');
}

await ap.locator('.ant-table-tbody .ant-table-row').first().getByRole('checkbox').check({ force: true });
await ap.waitForTimeout(300);
await shot(ap, '投票管理-批量操作条', ap.locator('.batch-toolbar, .ant-flex').filter({ hasText: '已选择' }).first());
const clearSel = ap.getByRole('button', { name: '取消选择' });
if (await clearSel.count()) await clearSel.click({ force: true });

await gotoHash(ap, '#/voting-v2/vote-v2-create', '新建活动');
await shot(ap, '新建活动-页面整体');
await shot(ap, '新建活动-手机预览', ap.locator('.vote-v2-phone, [class*="vote-v2-phone"]').first());
await shot(ap, '新建活动-基本设置', ap.locator('.ant-card').filter({ hasText: '1、基本设置' }).first());
await shot(ap, '新建活动-底部操作', ap.locator('.vote-v2-form-actions').first());

await ap.getByRole('tab', { name: '样式设置' }).click();
await ap.waitForTimeout(300);
await shot(ap, '新建活动-样式设置', ap.locator('.ant-card').filter({ hasText: '2、样式设置' }).first());

await ap.getByRole('tab', { name: '功能设置' }).click();
await ap.waitForTimeout(300);
await shot(ap, '新建活动-功能设置-分组', ap.locator('.ant-card').filter({ hasText: '投票分组' }).first());
await shot(ap, '新建活动-功能设置-参与范围', ap.locator('.ant-card').filter({ hasText: '参与范围' }).first());
await shot(ap, '新建活动-功能设置-投票规则', ap.locator('.ant-card').filter({ hasText: '投票规则设置' }).first());
await ap.getByText('按部门').click();
await ap.waitForTimeout(200);
await shot(ap, '新建活动-按部门选择', ap.locator('.ant-card').filter({ hasText: '参与范围' }).first());
await ap.getByText('全员').click();

await ap.getByRole('button', { name: '保存并发布' }).click();
await ap.waitForTimeout(400);
await ap.getByRole('tab', { name: '基本设置' }).click();
await ap.waitForTimeout(300);
await shot(ap, '新建活动-校验失败');

await gotoHash(ap, '#/voting-v2/vote-v2-edit/2', '编辑活动');
await shot(ap, '编辑活动-进行中页面', ap.locator('.page-stack').first());
await shot(ap, '编辑活动-进行中名称锁定', ap.locator('.ant-form-item').filter({ hasText: '活动名称' }).first());

await gotoHash(ap, '#/voting-v2/vote-v2-detail/2', '活动详情');
await shot(ap, '活动详情-页面整体');
await shot(ap, '活动详情-去编辑返回', ap.locator('.ant-space').filter({ hasText: '去编辑' }).first());
await shot(ap, '活动详情-页签', ap.locator('.ant-tabs-nav').first());

await gotoHash(ap, '#/voting-v2/vote-v2-detail/2/results', '投票结果');
await shot(ap, '活动详情-投票结果Tab');
await shot(ap, '活动详情-投票结果表格', ap.locator('.ant-table').first());

await gotoHash(ap, '#/voting-v2/vote-v2-detail/3/records', '投票记录');
await shot(ap, '活动详情-投票记录Tab');
await shot(ap, '活动详情-投票记录导出', ap.locator('.table-toolbar').first());

await gotoHash(ap, '#/voting-v2/vote-v2-edit/3', '编辑活动');
await shot(ap, '编辑活动-已结束只读', ap.locator('.vote-v2-form-actions').first());

await gotoHash(ap, '#/voting-v2/vote-v2-players/2', '选项管理');
await shot(ap, '选项管理-页面整体');
await shot(ap, '选项管理-查询筛选', ap.locator('.page-stack > *').filter({ hasText: '关键词' }).first());
await shot(ap, '选项管理-列表表格', ap.locator('.ant-table').first());
await shot(ap, '选项管理-工具栏', ap.locator('.list-table-card, .page-stack').locator('.ant-space').filter({ hasText: '添加选项' }).first());

await ap.getByRole('button', { name: '添加选项' }).click({ force: true });
try {
  await ap.locator('.ant-drawer').filter({ hasText: '选项标题' }).waitFor({ timeout: 8000 });
  await shot(ap, '选项管理-添加选项抽屉', ap.locator('.ant-drawer').filter({ hasText: '选项标题' }).last());
  await ap.keyboard.press('Escape');
  await ap.locator('.ant-drawer').filter({ hasText: '选项标题' }).waitFor({ state: 'hidden', timeout: 5000 }).catch(() => {});
} catch (err) {
  console.warn('add-drawer', err.message);
  await shot(ap, '选项管理-添加选项抽屉');
}

try {
  await ap.getByRole('button', { name: '详情 张工' }).first().click({ force: true });
  await ap.locator('.ant-drawer').waitFor({ timeout: 8000 });
  await shot(ap, '选项管理-选项详情抽屉', ap.locator('.ant-drawer').last());
  await ap.locator('.ant-drawer-extra, .ant-drawer-header').getByRole('button', { name: '编辑' }).click({ force: true });
  await ap.waitForTimeout(300);
  await shot(ap, '选项管理-编辑选项抽屉', ap.locator('.ant-drawer').last());
  await ap.keyboard.press('Escape');
  await ap.waitForTimeout(300);
} catch (err) {
  console.warn('player-drawer', err.message);
  await shot(ap, '选项管理-选项详情抽屉');
  await shot(ap, '选项管理-编辑选项抽屉');
}

try {
  await ap.getByRole('button', { name: '删除 张工' }).first().click({ force: true });
  await ap.getByText('删除选项').first().waitFor({ timeout: 5000 });
  await shot(ap, '选项管理-删除确认弹窗', ap.getByRole('dialog').last());
  await ap.keyboard.press('Escape');
} catch (err) {
  console.warn('player-delete', err.message);
  await shot(ap, '选项管理-删除确认弹窗');
}

try {
  await ap.getByRole('button', { name: '批量导入' }).click({ force: true });
  await ap.getByRole('dialog').waitFor({ timeout: 8000 });
  await shot(ap, '选项管理-批量导入弹窗', ap.getByRole('dialog').last());
  await shot(ap, '选项管理-批量导入表格', ap.getByRole('dialog').last());
  await ap.keyboard.press('Escape');
} catch (err) {
  console.warn('player-import', err.message);
  await shot(ap, '选项管理-批量导入弹窗');
  await shot(ap, '选项管理-批量导入表格');
}

await gotoHash(ap, '#/voting-v2/vote-v2-players/6', '选项管理');
await shot(ap, '选项管理-空状态', ap.locator('.ant-empty').first());

await ap.close();
await admin.close();

const h5 = await browser.newContext({ viewport: { width: 390, height: 844 }, deviceScaleFactor: 2, isMobile: true, hasTouch: true });
const hp = await h5.newPage();
hp.setDefaultTimeout(20000);

await gotoHash(hp, '#/c/h5/votes-v2', '投票');
await shot(hp, 'H5投票列表-进行中');
await shot(hp, 'H5投票列表-顶栏', hp.locator('.c-h5-top, .c-h5-header').first());
await shot(hp, 'H5投票列表-轮播', hp.locator('.c-home-banner, .home-banner, [class*="banner"]').first());
await shot(hp, 'H5投票列表-状态Tab', hp.locator('.c-tabs, .c-catalog-toolbar').first());
await shot(hp, 'H5投票列表-卡片', hp.locator('.c-h5-list, .c-h5-vote-card').first());

await hp.getByRole('button', { name: '未开始' }).click();
await hp.waitForTimeout(200);
await shot(hp, 'H5投票列表-未开始');

await hp.getByRole('button', { name: '已结束' }).click();
await hp.waitForTimeout(200);
await shot(hp, 'H5投票列表-已结束');

await gotoHash(hp, '#/c/h5/vote-v2-4', '食堂本周菜品');
await shot(hp, 'H5投票首页-单选进行中');
await shot(hp, 'H5投票首页-选手卡片', hp.locator('.vote-v2-phone-card, [class*="contestant"], .vote-v2-phone-grid').first());

try {
  await hp.getByRole('button', { name: '投票' }).first().tap();
  await hp.getByText('投票成功').waitFor({ timeout: 5000 });
  await shot(hp, 'H5投票成功弹窗', hp.locator('.c-vote-v2-success, [role="dialog"]').first());
  await hp.getByRole('button', { name: '确定' }).click({ force: true }).catch(() => {});
} catch (err) {
  console.warn('vote-success-miss', err.message);
  await shot(hp, 'H5投票成功弹窗');
}

await gotoHash(hp, '#/c/h5/vote-v2-2', '车间安全之星');
await shot(hp, 'H5投票首页-多选进行中');
await shot(hp, 'H5投票首页-多选已选底栏');

await gotoHash(hp, '#/c/h5/vote-v2-2/option-1', '张工');
await shot(hp, 'H5选项详情-页面整体');
await shot(hp, 'H5选项详情-底部投票', hp.locator('.c-h5-cta-bar').first());

await gotoHash(hp, '#/c/h5/vote-v2-3', '年度优秀作品展');
await shot(hp, 'H5投票首页-已结束');

await gotoHash(hp, '#/c/h5/vote-v2-1', '部门十佳员工评选');
await shot(hp, 'H5投票首页-未开始');

await gotoHash(hp, '#/c/h5/vote-v2-5', '评选活动');
await shot(hp, 'H5投票首页-不在范围内');

await gotoHash(hp, '#/c/h5/vote-v2-7', '静默结果页');
await shot(hp, 'H5投票首页-页面显示关闭');

await gotoHash(hp, '#/c/h5/votes-v2/mine', '我的投票记录');
await shot(hp, 'H5我的记录-列表');

await hp.close();
await h5.close();

const pc = await browser.newContext({ viewport: { width: 1280, height: 860 }, deviceScaleFactor: 1 });
const pp = await pc.newPage();
pp.setDefaultTimeout(20000);

await gotoHash(pp, '#/c/pc/votes-v2', '投票');
await shot(pp, 'PC投票列表-页面整体');
await shot(pp, 'PC投票列表-轮播', pp.locator('.c-home-banner, .home-banner, [class*="banner"]').first());
await shot(pp, 'PC投票列表-状态Tab', pp.locator('.c-catalog-toolbar, .c-tabs').first());
await shot(pp, 'PC投票列表-卡片网格', pp.locator('.c-pc-vote-grid').first());

await gotoHash(pp, '#/c/pc/vote-v2-4', '食堂本周菜品');
await shot(pp, 'PC投票首页-进行中');

await gotoHash(pp, '#/c/pc/vote-v2-2/option-1', '张工');
await shot(pp, 'PC选项详情-侧栏投票');
await shot(pp, 'PC选项详情-侧栏', pp.locator('.c-pc-side').first());

await gotoHash(pp, '#/c/pc/votes-v2/mine', '我的投票记录');
await shot(pp, 'PC我的记录-列表');

await gotoHash(pp, '#/c', 'C 端预览');
await shot(pp, 'C端门户-投票入口', pp.locator('.c-portal-main').first());

await pp.close();
await pc.close();
await browser.close();
console.log('done');

File: docs/prd/interest-groups/_capture.py
#!/usr/bin/env python3
"""Capture interest-group PRD screenshots from local Vite."""
from __future__ import annotations

from pathlib import Path
from playwright.sync_api import TimeoutError as PwTimeout
from playwright.sync_api import sync_playwright

ROOT = Path(__file__).resolve().parent / "screenshots"
ROOT.mkdir(parents=True, exist_ok=True)
BASE = "http://127.0.0.1:5173/"


def save(page, name: str, selector: str | None = None, full: bool = False) -> None:
    path = ROOT / f"{name}.png"
    try:
        if selector:
            loc = page.locator(selector).first
            loc.wait_for(state="visible", timeout=8000)
            loc.screenshot(path=str(path))
        else:
            page.screenshot(path=str(path), full_page=full)
        print("ok", name)
    except Exception as exc:  # noqa: BLE001
        print("fail", name, exc)


def goto(page, hash_path: str, wait: str = ".ant-layout, .c-ig-scroll, .c-portal, .c-signup-page") -> None:
    page.goto(BASE + hash_path, wait_until="domcontentloaded")
    page.wait_for_timeout(700)
    try:
        page.wait_for_selector(wait, timeout=12000)
    except PwTimeout:
        pass
    page.wait_for_timeout(400)


def click_text(page, text: str, exact: bool = False) -> None:
    loc = page.get_by_text(text, exact=exact).first
    loc.click(timeout=8000)
    page.wait_for_timeout(500)


def admin(page) -> None:
    page.set_viewport_size({"width": 1440, "height": 920})
    goto(page, "#/interest-groups")
    save(page, "后台-概览-页面整体")
    save(page, "后台-应用-左侧菜单", ".ant-layout-sider")
    save(page, "后台-概览-KPI指标卡", ".overview-kpi-grid")
    save(page, "后台-概览-分布图", ".overview-chart-row")
    save(page, "后台-概览-待办表", ".ant-table")

    goto(page, "#/interest-groups/interest-group-list")
    save(page, "后台-兴趣圈管理-页面整体")
    save(page, "后台-兴趣圈管理-查询筛选", ".search-panel, .list-search, form")
    save(page, "后台-兴趣圈管理-列表表格", ".ant-table")
    page.get_by_role("button", name="新建兴趣圈").click()
    page.wait_for_selector(".ant-drawer", timeout=8000)
    page.wait_for_timeout(400)
    save(page, "后台-兴趣圈管理-新建抽屉", ".ant-drawer")
    page.keyboard.press("Escape")
    page.wait_for_timeout(300)

    review = page.get_by_role("button", name="审核 午休飞盘局")
    if review.count() == 0:
        page.get_by_label("更多操作 午休飞盘局").click()
        page.wait_for_timeout(300)
        page.get_by_text("审核", exact=True).last.click()
    else:
        review.click()
    page.wait_for_selector(".ant-modal", timeout=8000)
    page.wait_for_timeout(300)
    save(page, "后台-兴趣圈管理-审核弹窗", ".ant-modal")
    page.keyboard.press("Escape")

    goto(page, "#/interest-groups/interest-group-detail/1")
    save(page, "后台-兴趣圈详情-页面整体")
    save(page, "后台-兴趣圈详情-头部卡片", ".activity-detail-header-card")
    save(page, "后台-兴趣圈详情-活动Tab")
    page.get_by_role("tab", name="成员").click()
    page.wait_for_timeout(500)
    save(page, "后台-兴趣圈详情-成员Tab")
    page.get_by_role("tab", name="评论").click()
    page.wait_for_timeout(500)
    save(page, "后台-兴趣圈详情-评论Tab")
    page.get_by_role("tab", name="精彩瞬间").click()
    page.wait_for_timeout(500)
    save(page, "后台-兴趣圈详情-瞬间Tab")

    goto(page, "#/interest-groups/interest-group-detail/5")
    save(page, "后台-兴趣圈详情-待审核状态")

    goto(page, "#/interest-groups/interest-group-activities")
    save(page, "后台-活动管理-页面整体")
    save(page, "后台-活动管理-查询筛选", ".search-panel, .list-search, form")
    save(page, "后台-活动管理-列表表格", ".ant-table")

    goto(page, "#/interest-groups/interest-group-activity-create")
    page.wait_for_timeout(600)
    save(page, "后台-新建活动-表单页", full=True)
    save(page, "后台-新建活动-基础信息")

    goto(page, "#/interest-groups/interest-group-activity-detail/101")
    save(page, "后台-活动详情-页面整体")
    save(page, "后台-活动详情-头部操作", ".activity-detail-header-card")
    page.get_by_role("tab", name="报名").click()
    page.wait_for_timeout(500)
    save(page, "后台-活动详情-报名Tab")
    page.get_by_role("tab", name="签到码").click()
    page.wait_for_timeout(500)
    save(page, "后台-活动详情-签到码Tab")
    page.get_by_role("tab", name="评论").click()
    page.wait_for_timeout(500)
    save(page, "后台-活动详情-评论Tab")
    page.get_by_role("tab", name="精彩瞬间").click()
    page.wait_for_timeout(500)
    save(page, "后台-活动详情-瞬间Tab")

    goto(page, "#/interest-groups/interest-group-activity-detail/104")
    save(page, "后台-活动详情-待审核活动")

    goto(page, "#/interest-groups/interest-group-categories")
    save(page, "后台-分类管理-页面整体")
    page.get_by_role("button", name="新建分类").click()
    page.wait_for_selector(".ant-modal", timeout=8000)
    page.wait_for_timeout(300)
    save(page, "后台-分类管理-新建弹窗", ".ant-modal")
    page.keyboard.press("Escape")

    goto(page, "#/interest-groups/interest-group-rules")
    save(page, "后台-规则设置-页面整体", full=True)
    save(page, "后台-规则设置-创建权限", ".ig-create-permission-card")
    save(page, "后台-规则设置-活动积分", ".activity-point-rules-card")


def h5_home(page) -> None:
    page.set_viewport_size({"width": 390, "height": 844})
    goto(page, "#/c/h5/interest-groups", ".c-ig-scroll")
    page.wait_for_timeout(800)
    save(page, "H5-首页-页面整体")
    save(page, "H5-首页-搜索入口", ".c-ig-search-wrap")
    save(page, "H5-首页-快捷入口", ".c-ig-shortcuts")
    save(page, "H5-首页-热门兴趣圈", ".c-ig-block.is-groups")
    save(page, "H5-首页-活动Tab", ".c-ig-tabs")
    save(page, "H5-首页-活动列表", ".c-ig-acts")


def click_shortcut(page, label: str) -> None:
    page.locator(".c-ig-shortcut", has_text=label).first.click()
    page.wait_for_timeout(600)


def h5_flows(page) -> None:
    page.set_viewport_size({"width": 390, "height": 844})

    goto(page, "#/c/h5/interest-groups", ".c-ig-scroll")
    page.locator(".c-ig-searchbar").click()
    page.wait_for_timeout(500)
    save(page, "H5-搜索-页面整体")
    page.locator(".c-ig-stack-back").click()
    page.wait_for_timeout(300)

    goto(page, "#/c/h5/interest-groups", ".c-ig-scroll")
    click_shortcut(page, "创建兴趣圈")
    save(page, "H5-创建兴趣圈-表单")
    save(page, "H5-创建兴趣圈-底部创建按钮", ".c-ig-form-bar")
    page.locator(".c-ig-stack-back").click()
    page.wait_for_timeout(300)

    goto(page, "#/c/h5/interest-groups", ".c-ig-scroll")
    click_shortcut(page, "创建活动")
    save(page, "H5-创建活动-表单", full=True)
    save(page, "H5-创建活动-底部创建按钮", ".c-ig-form-bar")
    page.locator(".c-ig-stack-back").click()

    goto(page, "#/c/h5/interest-groups", ".c-ig-scroll")
    click_shortcut(page, "我的活动")
    save(page, "H5-我的活动-列表")
    page.locator(".c-ig-stack-back").click()

    goto(page, "#/c/h5/interest-groups", ".c-ig-scroll")
    click_shortcut(page, "我的兴趣圈")
    save(page, "H5-我的兴趣圈-列表")
    page.locator(".c-ig-stack-back").click()

    goto(page, "#/c/h5/interest-groups", ".c-ig-scroll")
    page.locator(".c-ig-block.is-groups").get_by_text("全部", exact=True).click()
    page.wait_for_timeout(500)
    save(page, "H5-全部兴趣圈-列表")
    page.locator(".c-ig-stack-back").click()

    goto(page, "#/c/h5/interest-groups", ".c-ig-scroll")
    page.locator("section.c-ig-block").nth(1).get_by_text("全部", exact=True).click()
    page.wait_for_timeout(500)
    save(page, "H5-全部活动-列表")
    page.locator(".c-ig-stack-back").click()

    goto(page, "#/c/h5/interest-groups", ".c-ig-scroll")
    page.locator(".c-ig-hscroll .c-ig-gcard, .c-ig-hscroll button, .c-ig-hscroll a, .c-ig-hscroll > *").first.click()
    page.wait_for_timeout(700)
    save(page, "H5-兴趣圈详情-页面整体")
    save(page, "H5-兴趣圈详情-加入或退出", ".c-ig-group-panel")
    page.get_by_role("button", name="成员").click()
    page.wait_for_timeout(400)
    save(page, "H5-兴趣圈详情-成员Tab")
    page.get_by_role("button", name="圈子").click()
    page.wait_for_timeout(400)
    save(page, "H5-兴趣圈详情-圈子Tab")
    page.locator(".c-ig-float").click()
    page.wait_for_timeout(300)

    goto(page, "#/c/h5/interest-groups", ".c-ig-scroll")
    page.locator(".c-ig-acts li").first.click()
    page.wait_for_timeout(800)
    save(page, "H5-活动详情-页面整体", full=True)
    save(page, "H5-活动详情-底部报名", ".c-h5-detail-fab, .c-ig-cta, .c-ig-form-bar, footer")

    # pick enroll overlay if present
    if page.get_by_text("选择场次", exact=False).count() or page.locator(".c-ig-session").count():
        save(page, "H5-活动详情-场次选择")

    goto(page, "#/c/h5/ig-checkin/101", ".c-signup-page, .c-checkin-msg, .c-checkin-ok")
    page.wait_for_timeout(600)
    save(page, "H5-扫码签到-结果页")


def pc_flows(page) -> None:
    page.set_viewport_size({"width": 1440, "height": 920})
    goto(page, "#/c/pc/interest-groups", ".c-ig-scroll")
    page.wait_for_timeout(800)
    save(page, "PC-首页-页面整体")
    save(page, "PC-首页-快捷入口", ".c-ig-shortcuts")
    save(page, "PC-首页-活动三列", ".c-ig-acts")

    page.locator(".c-ig-searchbar").click()
    page.wait_for_timeout(500)
    save(page, "PC-搜索-页面整体")
    page.locator(".c-ig-stack-back").click()

    goto(page, "#/c/pc/interest-groups", ".c-ig-scroll")
    click_shortcut(page, "创建兴趣圈")
    save(page, "PC-创建兴趣圈-表单")
    page.locator(".c-ig-stack-back").click()

    goto(page, "#/c/pc/interest-groups", ".c-ig-scroll")
    click_shortcut(page, "创建活动")
    save(page, "PC-创建活动-表单", full=True)
    page.locator(".c-ig-stack-back").click()

    goto(page, "#/c/pc/interest-groups", ".c-ig-scroll")
    page.locator(".c-ig-acts li").first.click()
    page.wait_for_timeout(800)
    save(page, "PC-活动详情-页面整体")

    goto(page, "#/c/pc/interest-groups", ".c-ig-scroll")
    page.locator(".c-ig-hscroll").locator("xpath=./*").first.click()
    page.wait_for_timeout(700)
    save(page, "PC-兴趣圈详情-页面整体")


def extra_states(page) -> None:
    page.set_viewport_size({"width": 390, "height": 844})
    goto(page, "#/c/h5/interest-groups", ".c-ig-scroll")
    page.locator(".c-ig-searchbar").click()
    page.wait_for_timeout(400)
    page.locator("input").first.fill("zzzz无结果xyz")
    page.wait_for_timeout(400)
    save(page, "H5-搜索-空状态")

    page.set_viewport_size({"width": 1440, "height": 920})
    goto(page, "#/interest-groups/interest-group-list")
    page.get_by_placeholder("请输入兴趣圈名称").fill("zzzz无结果xyz")
    page.get_by_role("button", name="查询").click()
    page.wait_for_timeout(400)
    save(page, "后台-兴趣圈管理-空状态")


def main() -> None:
    with sync_playwright() as p:
        browser = p.chromium.launch(channel="chrome", headless=True)
        context = browser.new_context(device_scale_factor=1)
        page = context.new_page()
        page.set_default_timeout(10000)
        admin(page)
        h5_home(page)
        h5_flows(page)
        pc_flows(page)
        extra_states(page)
        browser.close()
    print("done", ROOT)


if __name__ == "__main__":
    main()
